src/coreclr/vm/interpexec.cpp

```diff
@@ -1233,6 +1234,139 @@ void InterpExecMethod(InterpreterFrame *pInterpreterFrame, InterpMethodContextFr
                     ip += 5;
                     break;
                 }
+                case INTOP_NEWARR:
+                {
+                    int32_t length = LOCAL_VAR(ip[2], int32_t);
+                    if (length < 0)
+                        assert(0); // Interpreter-TODO: Invalid array length
+
+                    CORINFO_CLASS_HANDLE arrayClsHnd = (CORINFO_CLASS_HANDLE)pMethod->pDataItems[ip[3]];
+                    size_t helperDirectOrIndirect = (size_t)pMethod->pDataItems[ip[4]];
+                    HELPER_FTN_NEWARR helper = nullptr;
+                    if (helperDirectOrIndirect & INTERP_INDIRECT_HELPER_TAG)
+                        helper = *(HELPER_FTN_NEWARR *)(helperDirectOrIndirect & ~INTERP_INDIRECT_HELPER_TAG);
+                    else
+                        helper = (HELPER_FTN_NEWARR)helperDirectOrIndirect;
+
+                    Object* arr = helper(arrayClsHnd, (intptr_t)length);
+                    LOCAL_VAR(ip[1], OBJECTREF) = ObjectToOBJECTREF(arr);
+
+                    ip += 5;
+                    break;
+                }
+#define LDELEM(dtype,etype)                                                    \
+do {                                                                           \
+    BASEARRAYREF arrayRef = LOCAL_VAR(ip[2], BASEARRAYREF);                    \
+    if (arrayRef == NULL)                                                      \
+        assert(0);                                                             \
+                                                                               \
+    ArrayBase* arr = (ArrayBase*)OBJECTREFToObject(arrayRef);                  \
+    uint32_t len = arr->GetNumComponents();                                    \
+    uint32_t idx = (uint32_t)LOCAL_VAR(ip[3], int32_t);                        \
+    if (idx >= len)                                                            \
+        assert(0);                                                             \
+                                                                               \
+    uint8_t* pData = arr->GetDataPtr();                                        \
+    size_t size = arr->GetComponentSize();                                     \
```

Comment:
Isn't this redundant ? Sounds like it should be known at compile time as `sizeof(etype)`

Reply:
Yes, thanks. Do you suggest taking size at compile time for perf reasons?

Reply:
`sizeof(etype)` is already a constant, no need to embed additional information. You already know the element size from the opcode type.